apps/HelloBaremetal/cmake-super_build/app/CMakeLists.txt

```diff
@@ -0,0 +1,51 @@
+cmake_minimum_required(VERSION 3.22)
+
+# Enable assembly language (.s) support additionally
+project(HelloBaremetal-app LANGUAGES C CXX ASM)
+
+if (NOT DEFINED GEN_EXE)
+    message(FATAL_ERROR "CMakeVariable GEN_EXE is not set")
+endif()
+
+# Set up C++ language settings
+set(CMAKE_CXX_STANDARD 17)
+set(CMAKE_CXX_STANDARD_REQUIRED YES)
+set(CMAKE_CXX_EXTENSIONS NO)
+
+if (BAREMETAL)
+    # Avoid handling Threads library as mandatory dependency in find_package()
+    set(Halide_NO_THREADS ON)
+    set(GEN_TARGET arm-32-noos-semihosting)
```

Comment:
These can be set in the toolchain file directly, rather than branching on `BAREMETAL`. Instead of `GEN_TARGET` use `Halide_TARGET`, which already exists.

Reply:
Good idea! I've set `Halide_TARGET` in toolchain file.
As for `Halide_NO_THREADS`, I will fix it accordingly once #7282 gets stable.